src/eponai/tools/cursive_arcadia_repl.clj

```diff
@@ -70,13 +90,13 @@
   [& args]
   (let [{:keys [arcadia-port port]} (:options (cli/parse-opts args cli-opts))]
     (print (format "Connecting to Arcadia on port %d... " arcadia-port))
-    (with-open [conn (nrepl/connect :port arcadia-port)]
+    (binding [*arcadia-port* arcadia-port]
+      (connect!)
```

Comment:
I wonder if we should isolate nREPL connection inside an object that handles the reconnection logic itself.

Something with a protocol like:
```clj
(defprotocol INREPLConn
  (connect! [this])
  (send-message! [this message]))
```

We'd get rid of one var and we could get rid of the `(binding [*arcadia-port* arcadia-port]` if the object is constructed with the port.

Does this make sense?